go2rtc_client/exceptions.py

```diff
@@ -2,6 +2,45 @@
 
 from __future__ import annotations
 
+from functools import wraps
+from typing import TYPE_CHECKING, Any
+
+from aiohttp import ClientError
+from mashumaro.exceptions import (
+    ExtraKeysError,
+    InvalidFieldValue,
+    MissingDiscriminatorError,
+    MissingField,
+    SuitableVariantNotFoundError,
+    UnserializableDataError,
+)
+
+if TYPE_CHECKING:
+    from collections.abc import Callable, Coroutine
+
 
 class Go2RtcClientError(Exception):
     """Base exception for go2rtc client."""
+
+
+def handle_error[**_P, _R](
+    func: Callable[_P, Coroutine[Any, Any, _R]],
+) -> Callable[_P, Coroutine[Any, Any, _R]]:
+    """Wrap aiohttp and mashumaro errors."""
+
+    @wraps(func)
+    async def _func(*args: _P.args, **kwargs: _P.kwargs) -> _R:
+        try:
+            return await func(*args, **kwargs)
+        except (
+            ClientError,
+            ExtraKeysError,
+            InvalidFieldValue,
+            MissingDiscriminatorError,
+            MissingField,
+            SuitableVariantNotFoundError,
+            UnserializableDataError,
+        ) as exc:
+            raise Go2RtcClientError from exc
```

Comment:
Should we include the original exception as an argument to the new exception or is that not needed when we raise from?

Reply:
It's not needed when we raise from: https://docs.python.org/3/library/exceptions.html#exception-context